contracts/liquidity_hub/pool-network/stableswap_3pool/src/helpers.rs

```diff
@@ -112,13 +149,32 @@ pub fn compute_offer_amount(
         .burn_fee
         .compute(before_commission_deduction.into());
 
-    Ok(OfferAmountComputation {
-        offer_amount,
-        spread_amount,
-        swap_fee_amount: swap_fee_amount.try_into()?,
-        protocol_fee_amount: protocol_fee_amount.try_into()?,
-        burn_fee_amount: burn_fee_amount.try_into()?,
-    })
+    #[cfg(not(feature = "osmosis"))]
+    {
+        Ok(OfferAmountComputation {
+            offer_amount,
+            spread_amount,
+            swap_fee_amount: swap_fee_amount.try_into()?,
+            protocol_fee_amount: protocol_fee_amount.try_into()?,
+            burn_fee_amount: burn_fee_amount.try_into()?,
+        })
+    }
+
+    #[cfg(feature = "osmosis")]
+    {
+        let osmosis_fee_amount = pool_fees
+            .osmosis_fee
+            .compute(before_commission_deduction.into());
+
+        Ok(OfferAmountComputation {
+            offer_amount,
+            spread_amount,
+            swap_fee_amount: swap_fee_amount.try_into()?,
```

Comment:
into() is used for infallible conversions, vs try_into() which can be fallible, thus returning a `Result`. In this case we are not 100% sure that we cane make the Uint256 into a Uint128, therefore we try and see if it's succeeds. It's just to be safe.

If we were converting Uint128 into Uint256 then we know for sure this will always succeed and we could use into().